cli/azd/pkg/workflow/config_test.go

```diff
@@ -0,0 +1,118 @@
+package workflow
+
+import (
+	"testing"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+var testWorkflow = &Workflow{
+	Name: "up",
+	Steps: []*Step{
+		{AzdCommand: Command{Args: []string{"package", "--all"}}},
+		{AzdCommand: Command{Args: []string{"provision"}}},
+		{AzdCommand: Command{Args: []string{"deploy", "--all"}}},
+	},
+}
+
+func Test_WorkflowMap_MarshalYAML(t *testing.T) {
+	workflowMap := WorkflowMap{}
+	workflowMap["up"] = testWorkflow
+	workflowMap["down"] = testWorkflow
+
+	expected, err := yaml.Marshal(workflowMap)
+	require.NoError(t, err)
+
+	actual, err := yaml.Marshal(workflowMap)
+	require.NoError(t, err)
+
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
+
+func Test_WorkflowMap_UnmarshallYAML(t *testing.T) {
+	t.Run("array style", func(t *testing.T) {
+		var wm WorkflowMap
+		yamlString := heredoc.Doc(`
+			up:
+			  - azd: package --all
+			  - azd: provision
+			  - azd: deploy --all
+		`)
+
+		err := yaml.Unmarshal([]byte(yamlString), &wm)
+		require.NoError(t, err)
+
+		upWorkflow, ok := wm["up"]
+		require.True(t, ok)
+
+		assertWorkflow(t, upWorkflow)
+	})
+
+	t.Run("map style", func(t *testing.T) {
+		var workflowMap WorkflowMap
+		yamlString := heredoc.Doc(`
+			up:
+			  steps:
+			    - azd: package --all
+			    - azd: provision
+			    - azd: deploy --all
+		`)
+
+		err := yaml.Unmarshal([]byte(yamlString), &workflowMap)
+		require.NoError(t, err)
+
+		upWorkflow, ok := workflowMap["up"]
+		require.True(t, ok)
+
+		assertWorkflow(t, upWorkflow)
+	})
+
+	t.Run("verbose style", func(t *testing.T) {
+		var workflowMap WorkflowMap
+		yamlString := heredoc.Doc(`
+			up:
+			  steps:
+			    - azd:
+			        command: package
```

Comment:
Still feels a little odd to me that we segement the `command` and `args` arrays when in the implementation we just combine "command" with "args".

I think I'd rather we just support these two forms:

```yaml
up:
  steps:
    - azd: provision --all # this does `strings.Split` on whitespace to build the slice we set as the args.
    - azd:
         args:    # here we just take the slice from the user, they have full control over it.
            - some
            - thing where I have explicit
            - control
            - over breaking
```

I don't see the value in breaking out the command into a separate thing?

Reply:
The command name just seems important enough to break out.  We could even offer better Intellisense here if we add the list of commands to the json schema.